### PR TITLE
Fix serialization issues

### LINKED ISSUE #1383: Serialisation
Hi,

Since few versions, it's not possible to serialize some objects like Predicate (and maybe OrderSpecicier).

I want to use the 3.6.3.

I'm using it to directly transfer restrictions between my FRONT and my BACK offices.

The class with problem is at least : com.mysema.query.types.path.PathBuilderValidator.
The 3 final anonymous classes inside aren't Serializable.

Thanks

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1383

### CODE REVIEW

**Comment by timowest** [comment]
I fixed the duplicate reflections dependency issue

**Comment by johnktims** [comment]
This looks fine to me. What do you think, @Shredder121 ?

File: querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java [hunk 1]
```diff
@@ -82,4 +82,15 @@ public FactoryExpressionBase(Class<? extends T> type) {
         return new FactoryExpressionWrapper<T>(this);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        } else if (o instanceof FactoryExpression) {
+            return getClass().equals(o.getClass()) && getArgs().equals(((FactoryExpression) o).getArgs());
```
Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java]: Does it sound reasonable to wrap this line before the `&&`?

**Comment by Shredder121** [comment]
To me it looks good as well, I will wait for answer to [this comment](https://github.com/querydsl/querydsl/pull/1386#discussion_r31389468) and I'll merge it.

File: querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java (hunk 1 above)
Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java]: Yes, that could be done.

Comment (Shredder121) [inline querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java]: Shall I do that then?

Comment (timowest) [inline querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java]: Ok. I'll continue working in the evening.

**Comment by Shredder121** [comment]
I updated both PRs.
@johnktims I think when the build finishes you can merge it if you agree.

**Pull request merged**